p2p/host.go

```diff
@@ -166,46 +152,36 @@ func (h *host) connect(addrInfo peer.AddrInfo) (err error) {
 	return err
 }
 
-// newStream opens a new stream with a specific peer using the host protocol
-func (h *host) newStream(p peer.ID) (network.Stream, error) {
+// send sends a non-status message to a specific peer
+func (h *host) send(p peer.ID, msg Message) (err error) {
 
 	// create new stream with host protocol id
-	stream, err := h.h.NewStream(h.ctx, p, h.protocolId)
+	s, err := h.h.NewStream(h.ctx, p, h.protocolId)
 	if err != nil {
-		return nil, err
+		log.Debug("Failed to create new stream", "err", err)
+		return err
```

Comment:
Avoid this, you're logging and returning an error. Where ever this is called probably catches that error and logs it too

Reply:
This should also be `log.Error`

Reply:
ok, will fix

Reply:
Ok, so this was actually intentional. This and the other uses of `log.Debug` in the `send` method return an error because the error is logged with `log.Error` when `send` returns an error, so these `Debug` logs are used to debug `send` method errors.

For example, if we attempted to send a message in the status submodule and `send` failed on `NewStream`, then we would get an `Error` log that says "Failed to send status message to peer" and a `Debug` log that says "Failed to create new stream".

Reply:
I think that is an interesting idea that perhaps we should explore more. 

Is it necessary to include the error in the log at every level?

Also, my expectation is that the error should be sufficient to provide context. "Failed to create new stream" as an error propagated to `status.sendMessages` seems sufficient given you are in the context of `sendMessages`. Would better error message perhaps solve this issue? Are there any cases of excessive propagation

Reply:
Is it necessary? I think so.

Without verbose mode the user would only see the error log. With verbose mode and in testing we see the error log and the debug log. Maybe I am misunderstanding how we use logs?

I think the context is and will become more important for debugging. Knowing that a status message failed to be sent (and what peer it was intended to be sent to) versus a broadcast message failed to be sent (and what peer it was intended to be sent to) would be helpful.

I think one level is not quite excessive but if two levels down would be.

Reply:
Will append information from lower level errors and pass up following discussion with David.

Reply:
Ok, removed debug logs for now. Error will be logged if `send` fails.